### PR TITLE
Larva burst now spawns the larva 20 seconds before burst, allowing the ghostrole to be taken

### PR DESCRIPTION
**Pull request by Dutch-VanDerLinde**
## About the PR
title
better version of #3215

## Why / Balance
so the larva that bursts near marines doesnt die instantly

also adds insane pain popups for when the larva is about to burst, for CM13 parity
image

## Technical details
spawns the larva in a container inside the host 20 seconds before it bursts, empties the container when they burst

## Media
ghostrole

https://github.com/user-attachments/assets/5f28a1d6-85b1-4691-83a3-65329ce3cbd7

popups

https://github.com/user-attachments/assets/b8b30383-4869-4b68-a53c-c3fdb24548e7

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**
:cl:
- tweak: The ghostrole for larvae will now appear 20 seconds before the host bursts

### CODE REVIEW

**Comment by Vermidia** [comment]
Is it possible to have it be 60 seconds before auto bursting? But the larva moving will instant burst the victim (that's how it is in CM-13)?

**Comment by Dutch-VanDerLinde** [comment]
> Is it possible to have it be 60 seconds before auto bursting? But the larva moving will instant burst the victim (that's how it is in CM-13)?

not sure how to do that and the larva is able to evolve inside their stomach

**Comment by Guess-My-Name** [comment]
Awesome change and looking forward to having it in the game,
i have to ask however -- in the edge case that a surgery happens ~10 seconds before the chest burst, what will happen?

**Comment by Dutch-VanDerLinde** [comment]
> Awesome change and looking forward to having it in the game, i have to ask however -- in the edge case that a surgery happens ~10 seconds before the chest burst, what will happen?

pretty sure you cant do surgery on a host that's about to burst, the surgery steps will appear greyed out

**Comment by UnicornOnLSD** [comment]
decent patch until the larva queue gets added

File: Content.Shared/_RMC14/Xenonids/Parasite/SharedXenoParasiteSystem.cs [hunk 1]
```diff
@@ -411,6 +411,18 @@ public override void Update(float frameTime)
             if (_net.IsClient)
                 continue;
 
+            // 20 seconds before burst, spawn the larva
+            if (infected.BurstAt - TimeSpan.FromSeconds(20) < time && infected.SpawnedLarva == null)
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/Xenonids/Parasite/SharedXenoParasiteSystem.cs:415]: The 20 seconds should be moved to a data field at some point but it's fine for now
```suggestion
            if (infected.BurstAt - TimeSpan.FromSeconds(20) < time && infected.SpawnedLarva == null)
```

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]
Need to check what 13 does but this will bug out if the surgery is done very late (in the last 20 seconds before burst)
The larva surgery only checks if the target has victim infected component

File: Content.Shared/_RMC14/Xenonids/Parasite/SharedXenoParasiteSystem.cs (hunk 1 above)
Comment (Dutch-VanDerLinde) [reply Content.Shared/_RMC14/Xenonids/Parasite/SharedXenoParasiteSystem.cs]: made this a datafield

**Comment by Dutch-VanDerLinde** [comment]
Fixed surgery breaking 20 seconds before burst and made it a datafield

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**